### ISSUE TITLE
switch all synchronous trial unit tests to using SynchronousTestCase

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5873]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5873|
|Type|enhancement|
|Created|2012-08-17 13:27:08Z|
|Branch|https://github.com/twisted/twisted/tree/sync-trial-tests-5873|

Many trial unit tests don't require the reactor spinning features of `twisted.trial.unittest.TestCase`.  To aid in porting trial to Python 3, make them use `SynchronousTestCase` instead, which will be ported before `TestCase` (thus allowing for easier test bootstrapping).

<details><summary>Searchable metadata</summary>

```
trac-id__5873 5873
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_sync_trial_tests_5873 branches/sync-trial-tests-5873
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1345210028000000 1345210028000000
changetime__1345729658000000 1345729658000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [35255]) Branching to 'sync-trial-tests-5873'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Okay.  I certainly switched a lot of them.  Hopefully I didn't miss any.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

Hooray for less reactor dependencies. There a few cases you may have missed that are left:

  1. Any reason you didn't do `weird.TestBleeding`? Note it doesn't return a Deferred.
  1. `test_test_visitor.py` looks like it could be switched too. 
  1. The various `InterruptedTest` instances in `test_keyboard.py`.
  1. `test_doctest` could be switched.
  1. `MockTest` in `test_shouldStop` in `test_runner`.

Once you change those, please merge. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

> Any reason you didn't do weird.TestBleeding? Note it doesn't return a Deferred.

I didn't want to think about it hard enough.  You're right, it can be switched.  Done in r35350.

> test_test_visitor.py looks like it could be switched too.

Visiting is abstractly synchronous, but I didn't port the visitor API to `SynchronousTestCase`.  It's deprecated so not worth it.

> The various InterruptedTest instances in test_keyboard.py.

Yes, but `SynchronousTestCase` does not handle C-c the same way as `TestCase` yet :/  I'm filing another  ticket for that, but I'm not sure it is terribly high priority.

> test_doctest could be switched.

Argh, but doctests... :(  Okay, I guess so.  Done in r35351.

> MockTest in test_shouldStop in test_runner.

There's three versions of this test, `DryRunTest`, `PyunitDryRunTest`, and `SynchronousDryRunTest`, each testing a different `TestCase` implementation.  So I think the one subclassing `twisted.trial.unittest.TestCase` should stay.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [35352]) Merge sync-trial-tests-5873

Author: exarkun
Reviewer: itamar
Fixes: [#5873](https://github.com/twisted/twisted/issues/5873)

Switch many of trial's own unit tests from `twisted.trial.unittest.TestCase`
to `twisted.trial.unittest.SynchronousTestCase`.  This removes reactor-related
code from the tests of any functionality that does not depend on the reactor.
